src/worker/tasks/nonceResyncWorker.ts

```diff
@@ -43,75 +44,46 @@ const handler: Processor<any, void, string> = async (job: Job<string>) => {
-    const { chainId, walletAddress } = splitSentNoncesKey(sentNonceKey);
+    try {
+      const { chainId, walletAddress } = splitSentNoncesKey(sentNonceKey);
 
-    const rpcRequest = getRpcClient({
-      client: thirdwebClient,
-      chain: await getChain(chainId),
-    });
-
-    const [transactionCount, lastUsedNonceDb] = await Promise.all([
-      eth_getTransactionCount(rpcRequest, {
-        address: walletAddress,
-        blockTag: "latest",
-      }),
-      inspectNonce(chainId, walletAddress),
-    ]);
+      const rpcRequest = getRpcClient({
+        client: thirdwebClient,
+        chain: await getChain(chainId),
+      });
+      const lastUsedNonceOnchain =
+        (await eth_getTransactionCount(rpcRequest, {
+          address: walletAddress,
+          blockTag: "latest",
+        })) - 1;
+      const lastUsedNonceDb = await inspectNonce(chainId, walletAddress);
 
-    if (Number.isNaN(transactionCount)) {
       job.log(
-        `Received invalid onchain transaction count for ${walletAddress}: ${transactionCount}`,
+        `wallet=${chainId}:${walletAddress} lastUsedNonceOnchain=${lastUsedNonceOnchain} lastUsedNonceDb=${lastUsedNonceDb}`,
       );
-      logger({
-        level: "error",
-        message: `[nonceResyncWorker] Received invalid onchain transaction count for ${walletAddress}: ${transactionCount}`,
-        service: "worker",
-      });
-      continue;
-    }
-
-    const lastUsedNonceOnchain = transactionCount - 1;
-
-    job.log(
-      `${walletAddress} last used onchain nonce: ${lastUsedNonceOnchain} and last used db nonce: ${lastUsedNonceDb}`,
-    );
-    logger({
-      level: "debug",
-      message: `[nonceResyncWorker] last used onchain nonce: ${transactionCount} and last used db nonce: ${lastUsedNonceDb}`,
-      service: "worker",
-    });
-
-    // If the last used nonce onchain is the same as or ahead of the last used nonce in the db,
-    // There is no need to resync the nonce.
-    if (lastUsedNonceOnchain >= lastUsedNonceDb) {
-      job.log(`No need to resync nonce for ${walletAddress}`);
       logger({
         level: "debug",
-        message: `[nonceResyncWorker] No need to resync nonce for ${walletAddress}`,
+        message: `[nonceResyncWorker] wallet=${chainId}:${walletAddress} lastUsedNonceOnchain=${lastUsedNonceOnchain} lastUsedNonceDb=${lastUsedNonceDb}`,
         service: "worker",
       });
-      continue;
-    }
 
-    //  for each nonce between last used db nonce and last used onchain nonce
-    //    check if nonce exists in nonce-sent set
-    //    if it does not exist, recycle it
-    for (
-      let _nonce = lastUsedNonceOnchain + 1;
-      _nonce < lastUsedNonceDb;
-      _nonce++
-    ) {
-      const exists = await isSentNonce(chainId, walletAddress, _nonce);
+      // Recycle all nonces between (onchain nonce, db nonce] if they aren't in-flight ("sent nonce").
+      for (
+        let nonce = lastUsedNonceOnchain + 1;
+        nonce <= lastUsedNonceDb;
```

Comment:
This is the only intended behavioral change: This worker should try to cancel up to lastUsedNonceDb inclusive, in case the most recent nonce is dropped/missing somehow.